src/server/_common.py

```diff
@@ -31,14 +31,29 @@ def _get_db() -> Connection:
 
 
 def get_real_ip_addr(req):  # `req` should be a Flask.request object
-    if REVERSE_PROXIED:
-        # NOTE: ONLY trust these headers if reverse proxied!!!
+    if REVERSE_PROXY_DEPTH:
+        # we only expect/trust (up to) "REVERSE_PROXY_DEPTH" number of proxies between this server and the outside world.
+        # a REVERSE_PROXY_DEPTH of 0 means not proxied, i.e. server is globally directly reachable.
+        # a negative proxy depth is a special case to trust the whole chain -- not generally recommended unless the
+        # most-external proxy is configured to disregard "X-Forwarded-For" from outside.
+        # really, ONLY trust the following headers if reverse proxied!!!
         if "X-Forwarded-For" in req.headers:
-            return req.headers["X-Forwarded-For"].split(",")[
-                0
-            ]  # take the first (or only) address from the comma-sep list
+            full_proxy_chain = req.headers["X-Forwarded-For"].split(",")
+            # eliminate any extra addresses at the front of this list, as they could be spoofed.
+            if REVERSE_PROXY_DEPTH > 0:
+                depth = REVERSE_PROXY_DEPTH
+            else:
+                # special case for -1/negative: setting `depth` to 0 will not strip any items from the chain
```

Comment:
```suggestion
                # special case for -1/0/negative: setting `depth` to 0 will not strip any items from the chain
```

Reply:
it cant have a value of zero here because of the `if REVERSE_PROXY_DEPTH:` above

Reply:
😱 

...that's probably fine